app/models/contact.rb

```diff
@@ -98,4 +98,16 @@ def self.contact_type_for(input_name)
   def phone_for_display
     phone.presence || cell_phone
   end
+
+  def client_opportunity_matches
+    @client_opportunity_matches ||= client_opportunity_match_contacts.joins(:match)
+  end
```

Comment:
This is equivalent to the relation: `client_opportunity_matches` above.  This is overriding the ActiveRecord method.  If we need different behavior, I'd suggest something like:
```suggestion
  has_many :matches, through: :client_opportunity_match_contacts, source: :match
```